src/speedCalculator.jsx

```diff
@@ -0,0 +1,49 @@
+import React, { useState, useEffect } from 'react';
+
+function SpeedCalculator(props) {
+  const [time, setTime] = useState(0);
+  const [startTime, setStartTime] = useState(0);
+  const [isActive, setIsActive] = useState(false);
+  useEffect(() => {
+    let interval;
+    if (isActive) {
+      setStartTime(Date.now() - time);
+      interval = setInterval(() => setTime(startTime ? Date.now() - startTime : 0), 100);
+    } else if (!isActive && time !== 0) {
+      clearInterval(interval);
+    }
+    return () => clearInterval(interval);
+  }, [isActive, startTime, time]);
+  function toggle() {
+    setIsActive(isActive => !isActive);
+  }
+  function reset() {
+    setTime(0);
+    setStartTime(0);
+  }
+  return (
+    <div className="col-12 my-4">
+      <hr/>
+      <h3>Test Your Reading Speed</h3>
+      <p>
+        Echidnas are medium-sized, solitary mammals covered with coarse hair
```

Comment:
This was a fun part of the app! (It's way beyond the requirements, but) you could even use an API to get a different piece of text on every page load. I'm sure there is an API for excerpts from public domain books. You can parse the response for the word count.